src/views/routes/users.js

```diff
@@ -61,16 +66,66 @@ router.post('/login', async (req, res) => {
   }
 });
 
+router.get('/:username', async (req, res) => {
+  try {
+    const { username } = req.params;
+    await getUser({ username });
+
+    responseHandler.handleSuccess(res, success.CODE);
+  } catch (error) {
+    const { message } = error;
+    errLogger.error(error);
+
+    if (message.includes(AUTH__EMAIL_DOES_NOT_EXIST)) {
+      responseHandler.handleError(res, notFound.CODE);
+    } else {
+      responseHandler.handleError(res, serverError.CODE);
+    }
+  }
+});
+
+router.post('/forgot-password', async (req, res) => {
+  try {
+    const { username } = req.body;
+
+    await sendRecoveryEmail(username);
+    responseHandler.handleSuccess(res, success.CODE);
+  } catch (error) {
+    const { message } = error;
+    errLogger.error(error);
+
+    if (message.includes(AUTH__EMAIL_DOES_NOT_EXIST)) {
+      responseHandler.handleError(res, badRequest.CODE);
+    } else {
+      responseHandler.handleError(res, serverError.CODE);
+    }
+  }
+});
 
-router.get('/:email', async (req, res) => {
+router.post('/reset-password', async (req, res) => {
   try {
-    const { email } = req.params;
-    const user = await getUser(email);
+    const { token, newPassword } = req.body;
```

Comment:
Although the token is random, an attacker with sufficient resources could spam this endpoint until he guesses a valid token, which would effectively reset a user's password.

You can bolster the security of this endpoint by requiring the username to be included in the request body. This way, the token and the matching username would have to be included with the new password in order to reset that user's password.